### PR TITLE
[FOGL-607] Payload Builder enhancements

### PR DESCRIPTION
**Pull request by amarendra-dianomic**

### CODE REVIEW

File: src/python/foglamp/storage/payload_builder.py [hunk 1]
```diff
@@ -74,22 +75,23 @@ def verify_orderby(arg):
     @classmethod
     def ALIAS(cls, *args):
         raise NotImplementedError("To be implemented")
-        return cls
 
     @classmethod
-    def SELECT(cls, *args):
+    def SELECT(cls, arg, *args):
```
[the remark this answers is not in the document]
Comment (ashish-ScaleDB) [reply src/python/foglamp/storage/payload_builder.py]: This needs to be an optional argument.
Also note that: for returning column names from storage layer in format be like `test->payload->where_test2_d1.json`. You may have to revisit this def.

File: src/python/foglamp/storage/payload_builder.py [hunk 2]
```diff
@@ -122,90 +127,132 @@ def DELETE(cls, tbl_name):
         return cls.FROM(tbl_name)
 
     @classmethod
-    def WHERE(cls, arg):
-        condition = {}
-        if cls.verify_condition(arg):
-            condition.update({"column": arg[0], "condition": arg[1], "value": arg[2]})
-            cls.query_payload.update({"where": condition})
+    def add_new_clause(cls, and_or, main, new):
+        """Recursive method to add 'new' to the last and/or of 'main'"""
```
Comment (ashish-ScaleDB) [inline src/python/foglamp/storage/payload_builder.py]: Can we have docstring here for `and_or, main, new`?

**Comment by amarendra-dianomic** [comment]
@ashish-ScaleDB all done, pl check.

File: src/python/tests/storage/data/payload_select_all.json [hunk 3]
```diff
@@ -1,3 +1 @@
-{
-  "columns": "*"
-}
\ No newline at end of file
+{}
```
Comment (ashish-ScaleDB) [inline src/python/tests/storage/data/payload_select_all.json]: Please delete this file

File: src/python/tests/storage/test_payload_builder.py [hunk 4]
```diff
@@ -129,21 +129,21 @@ def test_group_by_payload(self, test_input, expected):
...
         (["invalid", "values"], {})
     ])
     def test_aggregate_payload(self, test_input, expected):
         res = PayloadBuilder().AGGREGATE(test_input).payload()
         assert expected == json.loads(res)
 
     def test_select_all_payload(self):
-        res = PayloadBuilder().SELECT_ALL().payload()
+        res = PayloadBuilder().SELECT().payload()
         expected = _payload("data/payload_select_all.json")
```
Comment (ashish-ScaleDB) [inline src/python/tests/storage/test_payload_builder.py]: expected = {}

**Comment by amarendra-dianomic** [comment]
@ashish-ScaleDB pl check

**Pull request merged**